std_execution.bs

```diff
@@ -669,6 +669,58 @@ Sender consumers could be made pipeable, but we have chosen to not do so.
 However, since these are terminal nodes in a pipeline and nothing can be piped after them, we believe a pipe syntax may be confusing as well as unnecessary, as consumers cannot be chained.
 We believe sender consumers read better with function call syntax.
 
+## All awaitables are senders ## {#design-awaitables-are-senders}
+
+Since C++20 added coroutines to the standard, we expect that coroutines and awaitables will be how a great many will choose to express their asynchronous code. However, in this paper, we are proposing to add a suite of asynchronous algorithms that accept senders, not awaitables. One might wonder whether and how these algorithms will be accessible to those who choose coroutines instead of senders.
+
+In truth there will be no problem because all awaitable types automatically model the `typed_sender` concept. The adaptation is transparent and happens in the sender customization points, which are aware of awaitables.
+
+For an example, imagine a coroutine type called `task<T>` that knows nothing about senders. It doesn't implement any of the sender customization points. Despite that fact, and despite the fact that the `this_thread::sync_wait` algorithm is constrained with the `typed_sender` concept, the following would compile and do what the user wants:
+
+```c++
+task<int> doSomeAsyncWork();
+
+int main() {
+  // OK, awaitable types satisfy the requirements for typed senders:
+  auto o = this_thread::sync_wait(doSomeAsyncWork());
+}
+```
+
+Since awaitables are senders, writing a sender-based asynchronous algorithm is trivial if you have a coroutine task type: implement the algorithm as a coroutine. If you are not bothered by the possibility of allocations and indirections as a result of using coroutines, then there is no need to ever write a sender, a receiver, or an operation state.
+
+## Many senders can be trivially made awaitable ## {#design-senders-are-awaitable}
+
+If you choose to implement your sender-based algorithms as coroutines, you'll run into the issue of how to retrieve results from a passed-in sender. This is not a problem. If the coroutine type opts in to sender support -- trivial with the `execution::with_awaitable_senders` utility -- then a large class of senders are transparently awaitable from within the coroutine.
+
+For example, consider the following trivial implementation of the sender-based `retry` algorithm:
+
+<pre highlight="c++">
+template &lt;class S>
+  requires <i>single-typed-sender</i>&lt;S&> // See <a href="#spec-execution.coro_utils.as_awaitable">[execution.coro_utils.as_awaitable]</a>
+task&lt;<i>single-sender-value-type</i>&lt;S>> retry(S& s) {
```

Comment:
Should this be pass-by-value to avoid the dangling-reference problems?

This algorithm would be difficult to use in a sender composition expression as it won't accept temporary senders as arguments.